scripts/build.sh

```diff
@@ -3,110 +3,187 @@
 ################################################
 # Build script for sparkswapd
 #
-# Params:
-# - EXTERNAL_ADDRESS (optional, for hosted daemons)
+# Options:
+# -c, --no-cli                    do not copy certs for local cli installation
+# -d, --no-docker                 do not build docker images
+# -e=, --external-address=        your public IP address (removes prompt)
+#
 ################################################
 
-set -e -u
+set -eu
+
+echo "                                                   "
+echo "  ██████  ██▓███   ▄▄▄       ██▀███   ██ ▄█▀  ██████  █     █░ ▄▄▄       ██▓███  ";
+echo "▒██    ▒ ▓██░  ██▒▒████▄    ▓██ ▒ ██▒ ██▄█▒ ▒██    ▒ ▓█░ █ ░█░▒████▄    ▓██░  ██▒";
+echo "░ ▓██▄   ▓██░ ██▓▒▒██  ▀█▄  ▓██ ░▄█ ▒▓███▄░ ░ ▓██▄   ▒█░ █ ░█ ▒██  ▀█▄  ▓██░ ██▓▒";
+echo "  ▒   ██▒▒██▄█▓▒ ▒░██▄▄▄▄██ ▒██▀▀█▄  ▓██ █▄   ▒   ██▒░█░ █ ░█ ░██▄▄▄▄██ ▒██▄█▓▒ ▒";
+echo "▒██████▒▒▒██▒ ░  ░ ▓█   ▓██▒░██▓ ▒██▒▒██▒ █▄▒██████▒▒░░██▒██▓  ▓█   ▓██▒▒██▒ ░  ░";
+echo "▒ ▒▓▒ ▒ ░▒▓▒░ ░  ░ ▒▒   ▓▒█░░ ▒▓ ░▒▓░▒ ▒▒ ▓▒▒ ▒▓▒ ▒ ░░ ▓░▒ ▒   ▒▒   ▓▒█░▒▓▒░ ░  ░";
+echo "░ ░▒  ░ ░░▒ ░       ▒   ▒▒ ░  ░▒ ░ ▒░░ ░▒ ▒░░ ░▒  ░ ░  ▒ ░ ░    ▒   ▒▒ ░░▒ ░     ";
+echo "░  ░  ░  ░░         ░   ▒     ░░   ░ ░ ░░ ░ ░  ░  ░    ░   ░    ░   ▒   ░░       ";
+echo "      ░                 ░  ░   ░     ░  ░         ░      ░          ░  ░         ";
+echo "                                                                                 ";
 
 # Setting this env is ONLY required for a hosted broker setup.
 #
 # This address is used during the build process so that certs can be generated
 # correctly for a hosted (remote) broker daemon.
 EXTERNAL_ADDRESS=${EXTERNAL_ADDRESS:-}
-
-ARG=${1:-false}
+RELAYER_PROTO_VERSION='master'
+
+# parse options
+NO_CLI="false"
+NO_DOCKER="false"
+NO_CERTS="false"
+NO_IDENTITY="false"
+PRODUCTION="false"
+EXTERNAL_ADDRESS="localhost"
+for i in "$@"
+do
+case $i in
+    -c|--no-cli)
+    NO_CLI="true"
+
+    ;;
+    -d|--no-docker)
+    NO_DOCKER="true"
+
+    ;;
+    -e=*|--external-address=*)
+    EXTERNAL_ADDRESS="${i#*=}"
+
+    ;;
+    -i|--no-identity)
+    NO_IDENTITY="true"
+
+    ;;
+    -n|--no-certs)
+    NO_CERTS="true"
+
+    ;;
+    -p|--production)
+    PRODUCTION="true"
+
+    ;;
+    *)
+            # unknown option
+    ;;
+esac
+done
+
+if [ "$EXTERNAL_ADDRESS" == "" ]; then
+  echo "Please provide your public IP address"
+  read EXTERNAL_ADDRESS
+fi
 
 echo ""
 echo "It's time to BUILD! All resistance is futile."
 echo ""
 
-echo "Reinstalling dependencies"
-npm install
+if [ "$PRODUCTION" == "true" ]; then
+  echo "Reinstalling PRODUCTION dependencies"
+  npm install --quiet --production
```

Comment:
we don't install anything outside of production? I would think that we install but without the production flag

Reply:
@treygriffith the only time you would ever need to install dependencies is for tests. The user installs the broker-cli separately (and that installs the correct deps), so really... the only time we need `npm install` is for dev machines

Reply:
@treygriffith Maybe this isnt the correct way to go about it but i've omitted the installation line because it is redundant (you'll npm install locally, then npm install inside of the image)